src/interactive/Toaster.jsx

```diff
@@ -0,0 +1,136 @@
+import CSSTransitionGroup from 'react-addons-css-transition-group';
+import PropTypes from 'prop-types';
+import React from 'react';
+import cx from 'classnames';
+
+export const DELAY_TIME = 3000;
+
+/**
+ * @module Toaster
+ */
+class Toaster extends React.PureComponent {
+	constructor(props) {
+		super(props);
+
+		this.timeouts = [];
+		this.dismissedToasts = [];
+
+		this.cloneToast = this.cloneToast.bind(this);
+		this.setDismissedToast = this.setDismissedToast.bind(this);
+		this.clearTimeouts = this.clearTimeouts.bind(this);
+		this.setTimer = this.setTimer.bind(this);
+
+		this.state = {
+			toasts: this.props.toasts.map(this.cloneToast)
+		};
+	}
+
+	/**
+	 * @param {Object} dismissedToast - `Toast` components to remove from state
+	 * @returns undefined
+	 */
+	setDismissedToast(dismissedToast) {
+		this.setState({
+			toasts: this.state.toasts.filter(toast => dismissedToast.props.id !== toast.props.id)
+		});
+	}
+
+	/**
+	 * @returns undefined
+	 *
+	 * sets timers to delay each toast's dismissal
+	 */
+	setTimer() {
```

Comment:
I like the specificity of your `setDismissedToast` function that takes a `toast` as an argument - I think it would be nice to define this `setTimer` method the same way rather than building the `toast` looping into the method.

it could then be called like

```js
this.state.toasts.filter(t => t.props.autodismiss).map(this.setTimer);
```

I think a single-toast-at-a-time definition will also be required in order to handle new toasts coming in after mounting (you don't want to reset the timers on all the active toasts when a new one comes in)